src/main/java/org/kohsuke/github/GHRepository.java

```diff
@@ -873,10 +895,25 @@ public boolean hasAssignee(GHUser u) throws IOException {
      *             the io exception
      */
     public Set<String> getCollaboratorNames() throws IOException {
+        return getCollaboratorNames(CollaboratorAffiliation.ALL);
+    }
+
+    /**
+     * Gets the names of the collaborators on this repository. This method deviates from the principle of this library
+     * but it works a lot faster than {@link #getCollaborators()}.
+     *
+     * @param affiliation
+     *            Filter users by affiliation
+     * @return the collaborator names
+     * @throws IOException
+     *             the io exception
+     */
+    public Set<String> getCollaboratorNames(CollaboratorAffiliation affiliation) throws IOException {
         Set<String> r = new HashSet<>();
         // no initializer - we just want to the logins
         PagedIterable<GHUser> users = root.createRequest()
```

Comment:
This probably also use `listUsers(...)` right?

Reply:
`listUsers` does a few extra things, so that would negatively impact the speed of what is ostensibly the "fast" alternative. If that's a desirable change, I'm happy to make it.

Reply:
This isn't the fast alternative, just a convenient one. Go ahead and change it.

Reply:
Is the javadoc in error and I should update that while I'm in here?
```java
    /**
     * Gets the names of the collaborators on this repository. This method deviates from the principle of this library
     * but it works a lot faster than {@link #getCollaborators()}.
     *
     * @return the collaborator names
     * @throws IOException
     *             the io exception
     */
    public Set<String> getCollaboratorNames() throws IOException```